community/bolt/src/main/java/org/neo4j/bolt/security/auth/BasicAuthentication.java

```diff
@@ -62,17 +62,17 @@ public AuthenticationResult authenticate( Map<String,Object> authToken ) throws
         String password = safeCast( CREDENTIALS, authToken );
         if ( authToken.containsKey( NEW_CREDENTIALS ) )
         {
-            return update( user, password, safeCast( NEW_CREDENTIALS, authToken ) );
+            return update( user, password, safeCast( NEW_CREDENTIALS, authToken ), authToken );
         }
         else
         {
-            return authenticate( user, password );
+            return authenticate( user, password, authToken );
         }
     }
 
-    private AuthenticationResult authenticate( String user, String password ) throws AuthenticationException
+    private AuthenticationResult authenticate( String user, String password, Map<String,Object> authToken ) throws AuthenticationException
```

Comment:
Shouldn't it only take an `authToken`? In the basic case it will contain both the the username and password right?